src/app/pages/storage/snapshots/snapshot-table/snapshot-list.component.ts

```diff
@@ -0,0 +1,327 @@
+import helptext from 'app/helptext/storage/snapshots/snapshots';
+import { ConfirmOptions } from 'app/interfaces/dialog.interface';
+import { CoreEvent } from 'app/interfaces/events';
+import { WebsocketError } from 'app/interfaces/websocket-error.interface';
+import { ZfsSnapshot } from 'app/interfaces/zfs-snapshot.interface';
+import { EmptyConfig, EmptyType } from 'app/modules/entity/entity-empty/entity-empty.component';
+import { EntityToolbarComponent } from 'app/modules/entity/entity-toolbar/entity-toolbar.component';
+import { ToolbarConfig, ControlConfig } from 'app/modules/entity/entity-toolbar/models/control-config.interface';
+import { EntityUtils } from 'app/modules/entity/utils';
+import { SnapshotListRow } from 'app/pages/storage/snapshots/interfaces/snapshot-list-row.interface';
+import { SnapshotAddComponent } from 'app/pages/storage/snapshots/snapshot-add/snapshot-add.component';
+import { SnapshotCloneDialogComponent } from 'app/pages/storage/snapshots/snapshot-clone-dialog/snapshot-clone-dialog.component';
+import { SnapshotRollbackDialogComponent } from 'app/pages/storage/snapshots/snapshot-rollback-dialog/snapshot-rollback-dialog.component';
+import { SnapshotBatchDeleteDialogComponent } from 'app/pages/storage/snapshots/snapshot-table/components/snapshot-batch-delete-dialog/snapshot-batch-delete-dialog.component';
+import { loadSnapshots } from 'app/pages/storage/snapshots/store/snapshot.actions';
+import { selectSnapshotsTotal, SnapshotSlice } from 'app/pages/storage/snapshots/store/snapshot.selectors';
+import {
+  DialogService, ModalService, WebSocketService, AppLoaderService,
+} from 'app/services';
+import { CoreService } from 'app/services/core-service/core.service';
+import { SnapshotListEvent } from '../interfaces/snapshot-list-event.interface';
+import { snapshotChanged } from '../store/snapshot.actions';
+import { selectSnapshots, selectSnapshotState } from '../store/snapshot.selectors';
+
+@UntilDestroy()
+@Component({
+  templateUrl: './snapshot-list.component.html',
+  styleUrls: ['./snapshot-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [FormatDateTimePipe],
+})
+export class SnapshotListComponent implements OnInit, AfterViewInit {
+  error$ = this.store$.select(selectSnapshotState).pipe(map((state) => state.error));
+  isLoading$ = this.store$.select(selectSnapshotState).pipe(map((state) => state.isLoading));
+  isEmpty$ = this.store$.select(selectSnapshotsTotal).pipe(map((total) => total === 0));
+  @ViewChild(MatSort, { static: false }) sort: MatSort;
+  dataSource: MatTableDataSource<SnapshotListRow> = new MatTableDataSource([]);
+  defaultSort: Sort = { active: 'snapshot', direction: 'desc' };
+  filterString = '';
+  expandedRow: SnapshotListRow;
+  selection = new SelectionModel(true, []);
+  settingsEvent$: Subject<CoreEvent> = new Subject();
+  showExtraColumns$ = new BehaviorSubject(false);
```

Comment:
1. This setting belongs to a state. You should be reading it via selectors and updating it by dispatching actions.
2. This setting better belongs to user preferences, not local storage. Pull and see preferences state slice.